core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala

```diff
@@ -732,7 +735,12 @@ class FunctionPullingContainerProxy(
         data.action.fullyQualifiedName(false),
         data.action.rev,
         Some(data.clientProxy))
-
+    case Event(t: FailureMessage, data: WarmData) =>
+      logging.error(
+        this,
+        s"Failed to determine whether to keep or remove container on pause timeout for ${data.container.containerId}, retrying. Caused by: $t")
+      startSingleTimer(DetermineKeepContainer.toString, DetermineKeepContainer, 1.second)
```

Comment:
Is there any reason to start the timer after 1 second?
Since it will delay the deletion of the ETCD key for the problematic container, another request heading to this container can still come during that time.
The request will be rescheduled, but it will anyway also delay container creation.

Reply:
Hmm I think that shouldn't be the case. If a request comes in for the container, it will receive the `Initialize` message to unpause the container and go back to running. The retry message for `DetermineKeepContainer` after one second is cancelled at the beginning of the `Initialize` event along with the other timers so it should just gracefully unpause and go back to running so the latency is the duration of a normal unpause operation.

Reply:
And if for whatever reason a new request comes in and it can't unpause the container because the container is now broken for whatever reason, the activation will get rescheduled so there would be latency for that case which is already possible on any broken paused container but it's not any additional latency from the 1 second retry and the failure case of unpausing will be graceful now to correctly delete the broken container in all cases

Reply:
ok, I got confused and got another question.
If there is a problem in a connection with ETCD so `getLiveContainerCount` keeps failing, then isn't this container proxy run forever until the ETCD connection is restored?

Reply:
yes that's correct that's the current behavior I've set up. I guess in theory you could remove the container if the request fails, but you'd then break the guarantee of the keepingWarm count for the namespace

Reply:
Then how about introducing a maximum retry value?
After retrying x times to get the container count, if it fails, we can remove the container.
I feel it would be meaningless to keep the problematic containers for the keepingWarm count if the issue persists.

Reply:
I just changed it to retry a max of 5 times and added a test that the container deletion still occurs gracefully if the retries are exhausted. Been running this code for a few days and haven't had an issue so 100% sure now this fixes the issue. If it looks good to you now, I'll merge in the morning.